autogen/oai/client.py

```diff
@@ -588,6 +588,12 @@ def yes_or_no_filter(context, response):
                     raise
                 logger.debug(f"config {i} failed", exc_info=True)
                 if i == last:
+                    # Unfortunately, a timeout is not communicated in the error_code
```

Comment:
you can add a catch an openai `APITimeoutError` here, which inherits from `APIError` which is why it is caught here

Reply:
Ah that explains it, thank you!